addons/mail/migrations/12.0.1.0/end-migration.py

```diff
@@ -37,21 +37,25 @@ def fill_mail_thread_message_main_attachment_id(env):
     for model in thread_models:
         records = env[model].with_context(active_test=False).search(
             [('message_main_attachment_id', '=', False)])
-        grouped_attachments = env['ir.attachment'].read_group(
+        attachment_obj = env['ir.attachment'].with_context(
+            prefetch_fields=False,
+        )
+        grouped_attachments = attachment_obj.read_group(
             [('res_model', '=', model)],
-            fields=['res_id', 'id:array_agg'],
+            fields=['res_id'],
             groupby=['res_id'])
-        attachs = [(a['res_id'], a['id']) for a in grouped_attachments]
+        attachs = {
+            a['res_id']: attachment_obj.search(a['__domain'])
```

Comment:
shouldn't we add `.with_context(active_test=False)` in this `search`? 🤔

Reply:
Has attachment an `active` field?

Reply:
Yes

Reply:
https://github.com/OCA/OpenUpgrade/pull/1678

Reply:
attachment in Odoo 11 does not have `active` field.